File: 20220810/모범답안/20220810_07.js/Lamp.py
def solution(field, n):
    """
    :param field: int[][]
    :param n: int
    :return: int
    """

    fieldRow = len(field)
    fieldCol = len(field[0])
    rowSize = fieldRow - n + 1
    colSize = fieldCol - n + 1

    result = 0
    for row in range(rowSize):
        for col in range(colSize):
            result = max(result, countPlant(field, n, row, col))

    return result


def countPlant(field, n, row, col):
    plantCount = 0

    for curRow in range(row, row + n):
        for curCol in range(col, col + n):
            plantCount += field[curRow][curCol]

    return plantCount

File: 20220810/모범답안/20220810_07.js/Lamp.js
/**
 * @param field {number[][]}
 * @param n {number}
 * @return {number}
 */
function solution(field, n) {
  const fieldRow = field.length;
  const fieldCol = field[0].length;
  const rowSize = fieldRow - n + 1;
  const colSize = fieldCol - n + 1;

  let ret = 0;
  for (let row = 0; row < rowSize; row++) {
    for (let col = 0; col < colSize; col++) {
      ret = Math.max(ret, countPlant(field, n, row, col));
    }
  }

  return ret;
}

function countPlant(field, n, row, col) {
  let plantCount = 0;

  for (let curRow = row; curRow < row + n; curRow++) {
    for (let curCol = col; curCol < col + n; curCol++) {
      plantCount += field[curRow][curCol];
    }
  }

  return plantCount;
}

solution
